### DISCUSSION TITLE
[React-aria] — Compatibility with Next.js 13 app directory ?

### DISCUSSION THREAD

**Discussion by remib18** [adobe/react-spectrum#4471] [Q&A] [answered]
Hello there 👋🏻

I was trying to set up React-aria in a Next.js 13 app with the app directory architecture. However, I discovered that the `SSRProvider` does not seem to work in this architecture because it calls a client component as a server. Therefore, I attempted to set it up in a client component, but now I am receiving an error that it cannot import a helper.

I am not sure if I am misunderstanding how to set it up or if it is not compatible at the moment. If you are interested, I have provided detailed errors in this Stack Overflow question: https://stackoverflow.com/questions/76148697/react-aria-with-next-js-issues-with-ssrprovider-setup

Thank you in advance!

**Comment by LFDanLu** [comment]
What version of NextJS are you using? The `mjs` error you see with your latter two setups maybe a nextjs issue alluded to [here](https://github.com/adobe/react-spectrum/issues/4425#issuecomment-1518444687) where 13.2 works but 13.3 doesn't.

**Reply by remib18** [reply, marked as answer]
I've updated to nextJS 13.3.4, it seems to resolve the issue.

**Comment by kod-man** [comment]
Hey I have a question.  I am using Internationalization (i18n) in Server Components. My layout component is like this

```
import LanguageSwitcher from '@/components/common/LanguageSwitcher';
import { Inter } from 'next/font/google';
import { notFound } from 'next/navigation';

import React from 'react';
import { I18nProvider, SSRProvider, useLocale } from 'react-aria';
import './globals.css';

const inter = Inter({ subsets: ['latin'] });

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
};

export default function RootLayout({
  children,
  params,
}: {
  children: React.ReactNode;
  params: { locale: string };
}) {
  const { locale, direction } = useLocale();

  // Show a 404 error if the user requests an unknown locale
  if (params.locale !== locale) {
    notFound();
  }
  return (
    <SSRProvider>
      <I18nProvider locale={locale}>
        <html lang={locale} dir={direction}>
          <body className={inter.className}>
            <h1>Merhaba</h1>
            <LanguageSwitcher />
            {children}
          </body>
        </html>
      </I18nProvider>
    </SSRProvider>
  );
}
```

But I am getting an error. Am I doing something wrong? Version  "next": "13.4.4",

<img width="1080" alt="Screenshot 2023-06-12 at 22 26 37" src="https://github.com/adobe/react-spectrum/assets/54946794/fb13de55-4fe3-43a1-ad68-b9280d1c4cd5">

**Reply by devongovett** [reply]
You need "use client" in there.

**Reply by kod-man** [reply]
I added it but had to remove metadata first, then I got an error about notFound package. Since it is required for server-side-translation. I guess both don't work together


<img width="1035" alt="Screenshot 2023-06-13 at 09 25 34" src="https://github.com/adobe/react-spectrum/assets/54946794/b6127377-c08a-44c1-bc31-4392d1db1a7e">
